dotnet/src/Connectors/Connectors.AzureOpenAI/Services/AzureOpenAITextToImageService.cs

```diff
@@ -26,41 +28,115 @@ public class AzureOpenAITextToImageService : ITextToImageService
     /// <summary>
     /// Initializes a new instance of the <see cref="AzureOpenAITextToImageService"/> class.
     /// </summary>
-    /// <param name="modelId">The model to use for image generation.</param>
-    /// <param name="apiKey">OpenAI API key, see https://platform.openai.com/account/api-keys</param>
-    /// <param name="organizationId">OpenAI organization id. This is usually optional unless your account belongs to multiple organizations.</param>
-    /// <param name="endpoint">Non-default endpoint for the OpenAI API.</param>
+    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <param name="apiVersion">Azure OpenAI service API version, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
     public AzureOpenAITextToImageService(
-        string modelId,
-        string? apiKey = null,
-        string? organizationId = null,
-        Uri? endpoint = null,
+        string deploymentName,
+        string endpoint,
+        string apiKey,
+        string? modelId,
         HttpClient? httpClient = null,
-        ILoggerFactory? loggerFactory = null)
+        ILoggerFactory? loggerFactory = null,
+        string? apiVersion = null)
     {
-        this._client = new(modelId, apiKey, organizationId, endpoint, httpClient, loggerFactory?.CreateLogger(this.GetType()));
+        Verify.NotNullOrWhiteSpace(apiKey);
+
+        var connectorEndpoint = !string.IsNullOrWhiteSpace(endpoint) ? endpoint! : httpClient?.BaseAddress?.AbsoluteUri;
+        if (connectorEndpoint is null)
+        {
+            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+        }
+
+        var options = ClientCore.GetAzureOpenAIClientOptions(httpClient, apiVersion switch
```

Comment:
Why have the switch when we always hardcode the latest version?

Reply:
Right, the switch is no longer relevant and can be removed. It is just a leftover that was never taken out.